server/jetstream_cluster.go

```diff
@@ -4544,43 +4544,89 @@ func (cc *jetStreamCluster) remapStreamAssignment(sa *streamAssignment, removePe
 }
 
 type selectPeerError struct {
-	cluster      string
-	clusterPeers int
-	offline      int
-	excludeTag   int
-	noTagMatch   int
-	noStorage    int
-	uniqueTag    int
-	misc         int
+	excludeTag  bool
+	offline     bool
+	noStorage   bool
+	uniqueTag   bool
+	misc        bool
+	noJsClust   bool
+	noMatchTags map[string]struct{}
 }
 
 func (e *selectPeerError) Error() string {
-	return fmt.Sprintf(`peer selection cluster '%s' with %d peers
-offline: %d
-excludeTag: %d
-noTagMatch: %d
-noStorage: %d
-uniqueTag: %d
-misc: %d
-`,
-		e.cluster, e.clusterPeers, e.offline, e.excludeTag, e.noTagMatch, e.noStorage, e.uniqueTag, e.misc)
+	b := strings.Builder{}
+	b.WriteString("peer selection")
```

Comment:
Wonder if `peer select failures` is needed though, because we already return through NewJSClusterNoPeersError() which already has "no suitable peers for placement: "